# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoadHandler.h
# Commit: 7d5683180ac3
# Full Hash: 7d5683180ac3f9e7cc96d68dde27b8b48642a7ac
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r a48aff5f2555 -r 7d5683180ac3 dom/script/ScriptLoadHandler.h
--- a/dom/script/ScriptLoadHandler.h	Thu Feb 17 15:57:52 2022 +0000
+++ b/dom/script/ScriptLoadHandler.h	Thu Feb 17 15:57:52 2022 +0000
@@ -17,20 +17,23 @@
 #include "mozilla/RefPtr.h"
 #include "mozilla/UniquePtr.h"
 
+namespace JS::loader {
+class ScriptLoadRequest;
+}
+
 namespace mozilla {
 
 class Decoder;
 
 namespace dom {
 
-class ScriptLoadRequest;
 class ScriptLoader;
 class SRICheckDataVerifier;
 
 class ScriptLoadHandler final : public nsIIncrementalStreamLoaderObserver {
  public:
   explicit ScriptLoadHandler(
-      ScriptLoader* aScriptLoader, ScriptLoadRequest* aRequest,
+      ScriptLoader* aScriptLoader, JS::loader::ScriptLoadRequest* aRequest,
       UniquePtr<SRICheckDataVerifier>&& aSRIDataVerifier);
 
   NS_DECL_ISUPPORTS
@@ -96,7 +99,7 @@
   RefPtr<ScriptLoader> mScriptLoader;
 
   // The ScriptLoadRequest for this load. Decoded data are accumulated on it.
-  RefPtr<ScriptLoadRequest> mRequest;
+  RefPtr<JS::loader::ScriptLoadRequest> mRequest;
 
   // SRI data verifier.
   UniquePtr<SRICheckDataVerifier> mSRIDataVerifier;